### PR TITLE
Add documentation build test to CI/CD

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Add documentation build test to CI/CD.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -153,6 +153,30 @@ jobs:
...
+          python-version: '3.12'
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --requirement requirements/docs.txt
+          python -m pip install .
+
+      - name: Build docs
+        run: |
+          python -m sphinx -E -T -W -b html -D language=en -j auto -q docs/source docs/build
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yml:178]: It looks like we need to adjust `make doc` target and use it here instead.

**Review by arkid15r — COMMENTED** [review]
LGTM

File: .github/workflows/ci-cd.yml (hunk 1 above)
Comment (KJhellico) [reply .github/workflows/ci-cd.yml]: As I understand, this is actually the same command.

Comment (arkid15r) [reply .github/workflows/ci-cd.yml]: Cool, so we can just use it as is then.

**Review by arkid15r — APPROVED** [review]
👍

**Pull request merged**